### ISSUE TITLE
Defined hyperspectral-specific Transforms

### ISSUE THREAD

**Issue by nshaud** [nshaud/DeepHyperX#33]
`torchvision` defines [Transforms](https://pytorch.org/docs/stable/torchvision/transforms.html) objects to apply data augmentation and other transformations to data.

We could and should define our own custom Transforms.

Pros:
- easier to reuse the data augmentation transforms that we implement
- easier to apply various kinds of DA in the toolbox

Cons:
- adds an external dependency to `torchvision`

**Comment by mengxue-rs** [comment]
@nshaud Sorry bother you, **hyperspectral-specific Transforms** should include what kinds of transforms? do you mean dimension reduction methods like principle component transform and maximum noise fraction or data augmentation methods like radiation-based virtual samples and mixture-based virtual samples?

**Comment by nshaud** [comment]
Yes, I was mainly thinking data augmentation. Other augmentations such as dimension reduction could come later but I don't see any benefit to adding them now.
